datadreamer/prompt_generation/profanity_filter.py

```diff
@@ -0,0 +1,180 @@
+        self.device = device
+        self.use_lm = use_lm
+        if self.use_lm:
+            self.model, self.tokenizer = self._init_lang_model()
+
+    @staticmethod
+    def set_seed(seed: int) -> None:
+        """Sets the random seed for consistent prompt generation.
+
+        Args:
+            seed (int): The random seed.
+        """
+        random.seed(seed)
+        torch.manual_seed(seed)
+        torch.cuda.manual_seed_all(seed)
+
+    def _init_lang_model(self) -> Tuple[AutoModelForCausalLM, AutoTokenizer]:
+        """Initializes the language model and tokenizer for prompt generation.
+
+        Returns:
+            tuple: The initialized language model and tokenizer.
+        """
+        model_name = "Qwen/Qwen2.5-1.5B-Instruct"
+        model = AutoModelForCausalLM.from_pretrained(
+            model_name, torch_dtype="auto", device_map=self.device
+        )
+        tokenizer = AutoTokenizer.from_pretrained(model_name)
+        return model, tokenizer
+
+    def _contains_bad_words(self, texts: List[str]) -> bool:
+        """Checks if a list of texts contain bad words.
+
+        Args:
+            texts (List[str]): List of texts to checks against bad words list.
+
+        Returns:
+            bool: True if any of the texts contain bad words, False otherwise.
+        """
+        return any(text.lower() in BAD_WORDS_LIST for text in texts)
+
+    def _check_lm_safety(self, text: str) -> bool:
+        """Checks if a text is safe using a language model.
+
+        Args:
+            text (str): Text to check for bad words.
+
+        Returns:
+            bool: True if the text is safe, False otherwise.
+        """
+        if self.use_lm:
+            messages = [
+                {
+                    "role": "system",
+                    "content": self.LLM_PROMPT,
+                },
+                {"role": "user", "content": text},
+            ]
+            processed_text = self.tokenizer.apply_chat_template(
+                messages, tokenize=False, add_generation_prompt=True
+            )
+            model_inputs = self.tokenizer([processed_text], return_tensors="pt").to(
+                self.model.device
+            )
+            with torch.no_grad():
+                generated_ids = self.model.generate(
+                    **model_inputs,
+                    max_new_tokens=10,
+                    do_sample=False,
+                    top_k=None,
+                    top_p=None,
+                    temperature=None,
+                )
+                generated_ids = [
+                    output_ids[len(input_ids) :]
+                    for input_ids, output_ids in zip(
+                        model_inputs.input_ids, generated_ids
+                    )
+                ]
+                response = self.tokenizer.batch_decode(
+                    generated_ids, skip_special_tokens=True
+                )[0]
+                return response.lower().strip() == "appropriate"
```

Comment:
Can we expect the language model to always return the answer in exactly this form?